src/main.ts

```diff
@@ -1,10 +1,23 @@
-import { createApp } from "vue";
-import App from "./App.vue";
-import "./registerServiceWorker";
-import router from "./router";
-import store from "./store";
+import { createApp } from 'vue'
+import App from './App.vue'
+import './registerServiceWorker'
+import router from './router'
+import store from './store'
+import FirebaseApp from '@/../firebaseApp'
 
-createApp(App)
-  .use(store)
-  .use(router)
-  .mount("#app");
+import { User } from '@/models/User.ts'
+
+const init = () =>
+  createApp(App)
+    .use(store)
+    .use(router)
+    .mount('#app')
+
+FirebaseApp.auth().onAuthStateChanged(async (user: User) => {
+  if (user) {
+    // user commit
+  } else {
+    // another
+  }
+  init()
+})
```

Comment:
Depois a gente precisa conversar sobre essa parte, porque se a autenticação for para o back-end essa parte pode mudar um pouco

Reply:
Isso mesmo! Essa é a ideia! 😬 Enquanto o BE está sendo feito, manterei dessa forma caso o FE. hehehe